security/keyutils-tests.py

```diff
@@ -35,19 +35,33 @@ def setUp(self):
         for package in deps:
             if not smm.check_installed(package) and not smm.install(package):
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://git.kernel.org/pub/scm/linux/kernel/git/dhowells/keyutils.git"
-        git.get_repo(url, destination_dir=self.workdir)
-        os.environ['AUTOMATED'] = '1'
-        self.make_path = os.path.join(self.workdir, 'tests')
-        os.chdir(self.make_path)
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            def_url = ("https://git.kernel.org/pub/scm/linux/kernel/git/"
+                       "dhowells/keyutils.git")
+            url = self.params.get('url', default=def_url)
+            git.get_repo(url, destination_dir=self.workdir)
+            os.environ['AUTOMATED'] = '1'
```

Comment:
@nasastry is this not required for distro type?

Reply:
@Naresh-ibm nice catch, by make the above line common for 'upstream' and 'distro' ... distro test cases are executing properly.
Thanks!! Sending the updated patch.